extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunner.java

```diff
@@ -370,13 +370,15 @@ public Optional<ScalingStats> getScalingStats()
   @Override
   public Map<String, Long> getIdleTaskSlotCount()
   {
-    return Collections.emptyMap();
+    // A negative return value indicates tasks are queued in the thread pool
+    // but have not been scheduled to run in K8s yet.
+    return ImmutableMap.of(WORKER_CATEGORY, Long.valueOf(config.getCapacity() - tasks.size()));
   }
 
   @Override
   public Map<String, Long> getUsedTaskSlotCount()
   {
-    return Collections.emptyMap();
+    return ImmutableMap.of(WORKER_CATEGORY, Long.valueOf(Math.min(config.getCapacity(), tasks.size())));
```

Comment:
There can be a delay between tasks being added to the `tasks` map and when they are actually running. Can you explain how an operator should think about using this metric?

Reply:
As long as it's not over capacity, I think the delay is minimum (like within a second, we consider peon job submitted/start as taskSlot being used). So operator can use this metric to decide whether need to add more resource if it's always same as `config.getCapacity()`